src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts

```diff
@@ -0,0 +1,107 @@
+import { BackendSrv } from "@grafana/runtime";
+import { MockProxy } from "jest-mock-extended";
+import {
+  createFetchError,
+  createFetchResponse,
+  getQueryBuilder,
+  requestMatching,
+  setupDataSource,
+} from "test/fixtures";
+import { AssetCalibrationDataSource } from "./AssetCalibrationDataSource";
+import {
+  AssetCalibrationForecastGroupByType,
+  AssetCalibrationQuery,
+  CalibrationForecastResponse,
+} from "./types";
+
+let ds: AssetCalibrationDataSource, backendSrv: MockProxy<BackendSrv>
+
+beforeEach(() => {
+  [ds, backendSrv] = setupDataSource(AssetCalibrationDataSource);
+});
+
+const monthGroupCalibrationForecastResponseMock: CalibrationForecastResponse =
+{
+    calibrationForecast: {
+        columns: [
+            { name: "Month", values: ["2022-01-01T00:00:00.0000000Z", "2022-02-01T00:00:00.0000000Z", "2022-03-01T00:00:00.0000000Z"] },
+            { name: "Assets", values: [1, 0, 3] },
+        ]
+    }
+}
+
+const locationGroupCalibrationForecastResponseMock: CalibrationForecastResponse =
+{
+    calibrationForecast: {
+        columns: [
+          { name: "Location1", values: [1] },
+          { name: "Location2", values: [2] },
+          { name: "Location3", values: [3] },
+        ]
+    }
+}
```

Comment:
It's kind of weird that these response formats are different orientations. Is this API still in development? Why isn't the second response:

```js
calibrationForecast: {
    columns: [
      { name: "Location", values: ["Location 1", "Location 2", "Location 3"] },
      { name: "Assets", values: [1, 2, 3] }
    ]
}
```

Reply:
@mure we discussed if this should be an API level change or it should be a plugin implementation.

The way we are returning the data is the following:

calibrationForecast: {
    columns: [
      { name: "Location1", values: [1] },
      { name: "Location2", values: [2] }
      ....
    ]
}

To enable rendering the data on a bar chart, we want to ensure that the structure has the right format. Since this is a client specific need, we can implement it as part of the client code.

If you have concerns, please let me know and we can take it on a call

Reply:
@mure one more piece of detail that I have mentioned in the HLD:

The forecast can return 3 types of aggregations:

1. Time based 
    calibrationForecast: {
        columns: [
            { name: "Month", values: ["2022-01-01T00:00:00.0000000Z", "2022-02-01T00:00:00.0000000Z", "2022-03-01T00:00:00.0000000Z"] },
            { name: "Assets", values: [1, 0, 3] },
        ]
    }

2. Property based:

calibrationForecast: {
columns: [
{ name: "Location1", values: [1] },
{ name: "Location2", values: [2] }
....
]
}

3. Time & property based

calibrationForecast: {
columns: [
 { name: "Month", values: ["2022-01-01T00:00:00.0000000Z", "2022-02-01T00:00:00.0000000Z", "2022-03-01T00:00:00.0000000Z"] },
{ name: "Location1", values: [1, 2, 3] },
{ name: "Location2", values: [2, 3, 1] }
....
]
}


As you can see, the columns have a certain rule on how they are formatted. Having us provide a specific format from the backend only to satisfy a rendering option that the ui needs isn't something we want to do.